packages/fast-usdc/src/fast-usdc.contract.js

```diff
@@ -214,18 +214,26 @@ export const contract = async (zcf, privateArgs, zone, tools) => {
       privateArgs.assetInfo,
     );
   }
-
+  const nobleAccountV = zone.makeOnce('NobleAccount', () => makeNobleAccount());
   const feedKit = zone.makeOnce('Feed Kit', () => makeFeedKit());
 
   const poolAccountV = zone.makeOnce('PoolAccount', () => makeLocalAccount());
   const settleAccountV = zone.makeOnce('SettleAccount', () =>
     makeLocalAccount(),
   );
   // when() is OK here since this clearly resolves promptly.
-  /** @type {HostInterface<OrchestrationAccount<{chainId: 'agoric';}>>[]} */
-  const [poolAccount, settlementAccount] = await vowTools.when(
-    vowTools.all([poolAccountV, settleAccountV]),
+  /** @type {[HostInterface<OrchestrationAccount<{chainId: 'noble-1';}>>, HostInterface<OrchestrationAccount<{chainId: 'agoric';}>>, HostInterface<OrchestrationAccount<{chainId: 'agoric';}>>]} */
+  const [nobleAccount, poolAccount, settlementAccount] = await vowTools.when(
+    vowTools.all([nobleAccountV, poolAccountV, settleAccountV]),
+  );
+  trace('settlementAccount', settlementAccount);
+  trace('poolAccount', poolAccount);
+  trace('nobleAccount', nobleAccount);
+
+  const intermediateRecipient = await vowTools.when(
+    E(nobleAccount).getAddress(),
```

Comment:
what a pain that `.getAddress()` is async. Surely it's in the account exo state.

Reply:
Yes, `make(Cosmos|Local)Account` requires an address object for `initState` so this could be sync.

In some version of the world, I remember storing the ChainAddress as a vow in state. This might allow us to resolve `orch.makeAccount()` sooner. That would mean this still should be async.